tools/compare_codegen.sh

```diff
@@ -208,8 +209,14 @@ cleanup
 # Print mismatching files. Note that logs are expected to differ since timings are included
 set +e  # exit status of diff is 1 if there are any mismatches
 diffs=$(diff -qr -x '*.log' "$outdir/$origcommit" "$outdir/$comparecommit")
-echo -e "\n\nDIFF RESULT:\n$diffs\n\n"
-
-# Return number of mismatched cuda files. Success=0
-num_mismatches=$(echo "$diffs" | wc -l)
-exit "$num_mismatches"
+if [[ "$?" == 0 || -z "${diffs}" ]]
+then
+  echo "No difference found"
+  exit 0
+else
+  echo -e "\n\nDIFF RESULT:\n$diffs\n\n"
+
+  # Return number of mismatched cuda files. Success=0
+  num_mismatches=$(echo "$diffs" | wc -l)
+  exit "$num_mismatches"
+fi
```

Comment:
Now that I look at this, could we just simplify this to the following?
```c++
echo -e "\n\nDIFF RESULT:"
diff -qr -x '*.log' "$outdir/$origcommit" "$outdir/$comparecommit" && echo "No difference found"
exit $?
```

Reply:
Hmm, I think your script is correct, but let's keep it like the current way.

In the future, I'll need to parse the outputs and generate reports when there _are_ diffs and present those results to users, so it's necessary to not simplify the if-branch for now.

Echoing the diff lines is also part of the "output".

Reply:
Actually I think you're right, I missed that you put `diff -qr ...` thing outside of $()